### ISSUE TITLE
Autopilot fail opening channels

### ISSUE THREAD

**Issue by k0dlaty** [lightninglabs/lightning-app#1074]
Autopilot fail to open channels

It seems that there is some problem with calculation
"BTC: not enough witness outputs to create funding transaction, need 0.00303704 BTC only have 0.00300149 BTC  available"

Part of a log.

lnd: 2019-04-25 00:12:27.249 [INF] SRVR: Established connection to: 02c91d6aa51aa940608b497b6beebcb1aec05be3c47704b682b3889424679ca490@213.174.156.65:9735
2019-04-25 00:12:27.249 [INF] SRVR: Finalizing connection to 02c91d6aa51aa940608b497b6beebcb1aec05be3c47704b682b3889424679ca490@213.174.156.65:9735, inbound=false
lnd: 2019-04-25 00:12:27.249 [INF] FNDG: Initiating fundingRequest(localAmt=0.00285141 BTC, remoteAmt=0 mSAT, capacity=0.00285141 BTC, chainhash=000000000019d6689c085ae165831e934ff763ae46a2a6c172b3f1b60a8ce26f, peer=02c91d6aa51aa940608b497b6beebcb1aec05be3c47704b682b3889424679ca490, dustLimit=0.00000573 BTC, min_confs=0)
2019-04-25 00:12:27.249 [INF] LNWL: Performing funding tx coin selection using 26406 sat/kw as fee rate
2019-04-25 00:12:27.249 [WRN] ATPL: Unable to open channel to 02c91d6aa51aa940608b497b6beebcb1aec05be3c47704b682b3889424679ca490 of 0.00285141 BTC: not enough witness outputs to create funding transaction, need 0.00303704 BTC only have 0.00300149 BTC  available
2019-04-25 00:12:27.249 [INF] SRVR: Disconnecting from 02c91d6aa51aa940608b497b6beebcb1aec05be3c47704b682b3889424679ca490@213.174.156.65:9735
2019-04-25 00:12:27.249 [INF] PEER: Disconnecting 02c91d6aa51aa940608b497b6beebcb1aec05be3c47704b682b3889424679ca490@213.174.156.65:9735, reason: server: disconnecting peer 02c91d6aa51aa940608b497b6beebcb1aec05be3c47704b682b3889424679ca490@213.174.156.65:9735
lnd: 2019-04-25 00:12:28.590 [INF] ATPL: Attempting to execute channel attachment directives: (map[autopilot.NodeID]*autopilot.AttachmentDirective) (len=1) {
 (autopilot.NodeID) (len=33 cap=33) {
  00000000  02 6b e7 57 81 a3 5c a2  27 32 20 0f 5b c8 2a 1e  |.k.W..\.'2 .[.*.|
  00000010  8a 21 24 81 54 e9 50 a1  54 8b ab 0c dd 44 d2 ec  |.!$.T.P.T....D..|
  00000020  20                                                | |
 }: (*autopilot.AttachmentDirective)(0xc00007a5a0)({
  NodeID: (autopilot.NodeID) (len=33 cap=33) {
   00000000  02 6b e7 57 81 a3 5c a2  27 32 20 0f 5b c8 2a 1e  |.k.W..\.'2 .[.*.|
   00000010  8a 21 24 81 54 e9 50 a1  54 8b ab 0c dd 44 d2 ec  |.!$.T.P.T....D..|
   00000020  20                                                | |
  },
  ChanAmt: (btcutil.Amount) 0.00285141 BTC,
  Addrs: ([]net.Addr) (len=1 cap=1) {
   (*net.TCPAddr)(0xc001d1eba0)(93.147.136.98:9735)
  }
 })
}

**Comment by rjbiii** [comment]
Same issue here. It always seems to want just more than what my available balance is.

2019-04-24 18:05:19.672 [WRN] ATPL: Unable to open channel to 02247d9db0dfafea745ef8c9e161eb322f73ac3f8858d8730b6fd97254747ce76b of 0.00361 BTC: not enough witness outputs to create funding transaction, need 0.00389234 BTC only have 0.0038 BTC  available

**Comment by halseth** [comment]
lnd issue: https://github.com/lightningnetwork/lnd/issues/986

**Comment by rjbiii** [comment]
Follow up:  I was running from the AppImage without integrating with the OS (ubuntu 18.04).  I ran it again after integrating with OS and now it's opening channels.

**Comment by mookfish2004** [comment]
Hey All,
            Sorry that i'm crashing the party but tried to create my own post but was not successful and this post seemed to the closest to my question.

Q) :- If you activate Autopilot and gain some channels can you deactivate autopilot without effecting the channels you have gained?
 the reason I ask is because I started Autopilot with the idea of just gaining a few channels to improve my capacity on the network, I have since disabled Autopilot and now all my channels including ones I had connected to without Autopilot are now showing as Inactive.
 Do I just need to wait, Do I need to reactivate Autopilot?
I turned Autopilot off as I had gained enough channels and wanted to preserve the rest of my bank.
Help with this matter would be most appreciated.
Many Thanks.

**Comment by halseth** [comment]
@mookfish2004 That is correct, autopilot only opens channels, so you can turn it off when you have enough channels :)

**Comment by tanx** [comment]
@halseth I've been testing the mainnet build on iOS and it looks like the PR fixes this issue.

**Comment by tanx** [comment]
Looks like this issue is solved on mobile with the `mobile-autopilot-100` branch. @halseth any ETA on merging the fix in lnd master? Then we can close this issue for desktop as well.

**Comment by TheRec** [comment]
Could someone address the questions from @wdey in #1095, I've funded a wallet on Android today for ~18000 sats and I have exactly the same feedback, autopilot does not seem to work and there is no way to tell from the UI what the issue is.

If there's minimum for autopilot it should be more clear, especially if it's intended to be a feature that does not require much user intervention.

**Comment by halseth** [comment]
@tanx The fix is now merged to master.
